accepted/0000-link-packages-to-source-and-build.md

```diff
@@ -0,0 +1,514 @@
+
+The entry will be stored and retrieved using the shasum of the package tarball. This will make sure we don't find any signatures for a  deleted package that has been resurrected with the same name and version.
+
+There will be no way of deleting an entry if this was published unintentionally to the ledger as it's immutable by design to prevent tampering. The practical consequence of publishing to the ledger is that information about your package (name, version), repository (commit, branch, owner) and workflow (name, id) are uploaded to the public ledger. A possible workaround would be to rename the leaked repo, make it private and delete the old one.
+
+##### Detailed steps to publish
+Interaction between the npm CLI tool running in CI, the npm registry, and the Sigstore components Fulcio and Rekor:
+
+```mermaid
+  sequenceDiagram
+    participant npm_cli
+    participant fulcio
+    participant rekor
+    participant npm_registry
+    npm_cli->>fulcio: Request signing certificate
+    fulcio->>npm_cli: Signing certificate
+    npm_cli->>rekor: Upload signed build provenance attestation
+    rekor->>npm_cli: Rekor entry
+    npm_cli->>npm_registry: Publish package and upload Rekor entry
+    npm_registry->>fulcio: Request signing certificate
+    fulcio->>npm_registry: Signing certificate
+    npm_registry->>rekor: Upload signed release attestations
+    rekor->>npm_registry: Rekor entry
+    npm_registry->>npm_registry: Attach rekor entry to published package
+    npm_registry->>npm_cli: Success
+```
+
+1. `npm publish --build-provenance` detects execution within a supported CI/CD environment and initializes token exchange to get a JWT id token from the CI/CD system's OIDC identity provider.
+2. npm CLI creates a new ephemeral key pair, requests a certificate from Fulcio (CA) and authenticates via the JWT id token.
+3. A SLSA build provenance attestation is created based on information available within the JWT and CI system and signs it with the ephemeral key.
+4. npm CLI uploads the completed attestation record on Rekor for public packages.
+5. npm registry creates a release attestation, stating that the version has been accepted and published, and places the record on Rekor for public packages.
+
+#### Verifying
+The package integrity captured by the provenance attestation will be verified by the npm CLI. To begin with, the package integrity should be verified when running the  `npm audit signatures` command and eventually transparently integrated into `npm install` and enabled by default.
+
+This will be staged rollout over several npm CLI releases as changing the default behavior of `npm install` is a breaking change:
+
+1. The existing `npm audit signatures` command will start verifying build signatures, possibly adding an opt-in flag to audit these signatures during `npm install`, e.g. `--audit-build-provenance`
```

Comment:
rather than forever adding standalone flags, it'd be nice to think about the long-term DX of flags. Adding `npm install --audit=build-provenance` rather than a single flag would be preferable IMO.

Reply:
> `npm install --audit=build-provenance`

Nice! This looks preferable to me to 👍